_posts/2020-02-03-ftp.md

```diff
@@ -0,0 +1,80 @@
+
+FTP now is a fairly redundant mode of file transfer. It is inefficient, and although it has been updated somewhat over the years, it is still an old protocol designed for non-graphical systems. The problem lies in the fact that many applications still use FTP today, despite support being phased out by various distributors including the [Debian Project](https://www.debian.org/News/2017/20170425). 
+
+But why is it insecure? On the surface, it looks as though FTP can be secured through username and password. However, this isn't really the case, or rather it's not strictly speaking secure. This is in part because FTP doesn't actually *encrypt* its data. Any authentication data is transferred as a plaintext file making it particularly vulnerable to spoofing. In other words, an attacker could 'listen' to your packets, gaining information regarding your login credentials. This is because of the plaintext format, your username and password can be easily read and used to breach your system. Beyond this as well is the fact that FTP is no longer maintained, it has not actually been updated to be compatible with the modern world.
+
+This begs the question: why is FTP still used? Primarily because it's good for large quantities of data and is easily automated. but mostly because it is free and open source. In instances where security isn't a priority, FTP can be good enough, but unfortunately it is still being used in instances where data protection should be of [utmost importance](https://www.healthcareitnews.com/news/205000-patient-records-exposed-misconfigured-ftp-server). 
+Fortunately, there is growing movement away from FTP servers, especially as support is discontinued by many vendors and organizations. That isn't to say that FTP is dying, just that it is moving away from using TCP as its protocol of choice when interfacing with the internet. 
+
+## Alternatives to FTP 
+
+There are actually several methods to transfer files over the internet, including HTTPS and AS2, however as alternatives to FTP there are really only 2 of interest: SFTP and FTPS. Both are known as 'Secure FTP'
+
+### SFTP
+
+SSH File Transfer Protocol (SFTP) encompasses the same principle as FTP in that it is a file transfer protocol, but uses a secure connection instead. 
+SFTP uses the the SSH protocol to secure traffic and listens on port 22 because that is the port that SSH is typically configured to. It is possible to reconfigure SFTP to listen on different ports, but arguably one of its advantages that it's easier to protect via firewalls due to the use of a single port. 
+To put it simply, SFTP is secure because SSH is secure. It is harder to spoof data packets as they are not contained in plaintext files and commands are kept separate to the file it is associated to leading to faster connections and making it harder to intercept.
```

Comment:
Probably also worth mentioning here that OpenSSH has had likely had far more security scrutiny than any other ftp or other file transfer protocol due to the fact that SSH itself is used everywhere.